include/pybind11/stl.h

```diff
@@ -67,22 +67,59 @@ struct set_caster {
     }
     void reserve_maybe(const anyset &, void *) {}
 
+    template <typename T = Type, enable_if_t<has_reserve_method<T>::value, int> = 0>
+    void reserve_maybe(const sequence &s, Type *) {
+        value.reserve(s.size());
+    }
+    void reserve_maybe(const sequence &, void *) {}
+
+    template <typename ContainerType>
+    bool insert_elements(const ContainerType &container, bool convert) {
```

Comment:
Could we move this to the `private:` section?

Reply:
Looks like it is already in the private section. I moved the other `insert_elements` to private.

Reply:
oh, sorry, it's so difficult to see in the GH view